### ISSUE TITLE
unified interface for devices that can be triggered

### ISSUE THREAD

**Issue by carandraug** [MicronOxford/microscope#159]
We have `TriggerTargetMixin` for devices that can be triggered.  Currently, it is only used by deformable mirrors and the ximea camera but makes sense to use on all cameras and any other devices types can are typically triggered. This was discussed a few times before but seems like we never opened an issue for this.  Basically, the goal is:

1. have all triggerable devices share the same interface and logic
2. extend the existing trigger modes in the camera interface to cover other common trigger types and modes

**Comment by iandobbie** [comment]
Yes I guess that the triggering should be generalised to cover devices that need this type of logic to be controlled.
